### DISCUSSION TITLE
How to pass parameters (hv_param) to a PureDataStream

### DISCUSSION THREAD

**Discussion by jonascant** [pschatzmann/arduino-audio-tools#1638] [Q&A] [answered]
I would like to use values generated by a rotary encoder to manipulate objects in a puredata patch. I think I can use @hv_param, but i don't see a way to pass it in the arduino sketch.

**Comment by pschatzmann** [comment, marked as answer]
I never tried anyhting like this, but I suggest to look into the sendFloatToReceiver method of the HeavyContext object.
```C
  /**
   * A convenience function to send a float to a receiver to be processed immediately.
   * The receiver is addressed with its hash, which can also be determined using hv_stringToHash().
   * This function is thread-safe.
   *
   * @return  True if the message was accepted. False if the message could not fit onto
   *          the message queue to be processed this block.
   */
  virtual bool sendFloatToReceiver(hv_uint32_t receiverHash, float f) = 0;
```

**Reply by simmit123456789** [reply]
if anyone has figured this out I'd love to know :) in a similar way I think it should also be possible to pass midi in to control things in the pd patch but I'm as yet unable to get it working

**Reply by simmit123456789** [reply]
https://github.com/Wasted-Audio/hvcc/blob/develop/docs/04.midi.md
